Subject: xmlstarlet: fix linking
From: Michael Olbrich <m.olbrich@pengutronix.de>

xmlstarlet links libexslt statically. The configure script does not check for
the dependencies. This fails if libexslt links agains libgcrypt.

This patch changes the configure script, so that all libs are linked
dynamically. The linker does the rest.

For ptxdist this is ok. We install the libs already.
Not for upstream.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 configure.in |   16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

Index: b/configure.in
===================================================================
--- a/configure.in
+++ b/configure.in
@@ -31,7 +31,7 @@ AC_ARG_WITH(libxml-prefix,
         [  --with-libxml-prefix=[PFX]		Specify location of libxml],
         LIBXML_PREFIX=$withval
         LIBXML_CFLAGS="-I$withval/include/libxml2 -I$withval/include/libxml"
-        LIBXML_LIBS="$withval/lib/libxml2.a -lz -lm -lpthread -ldl"
+        LIBXML_LIBS="$withval/lib/libxml2.so -lz -lm -lpthread -ldl"
 )
         
 AC_ARG_WITH(libxml-include-prefix,
@@ -41,13 +41,13 @@ AC_ARG_WITH(libxml-include-prefix,
 
 AC_ARG_WITH(libxml-libs-prefix,
         [  --with-libxml-libs-prefix=[PFX]	Specify location of libxml libs],
-        LIBXML_LIBS="$withval/libxml2.a -lz -lm -lpthread -ldl"
+        LIBXML_LIBS="$withval/libxml2.so -lz -lm -lpthread -ldl"
 )
 
 AC_ARG_WITH(libxml-src,
         [  --with-libxml-src=[DIR]               For libxml thats not installed yet (sets all three above)],
         LIBXML_SRC="$withval"
-        LIBXML_LIBS="$withval/.libs/libxml2.a -lz -lm -lpthread"
+        LIBXML_LIBS="$withval/.libs/libxml2.so -lz -lm -lpthread"
         LIBXML_CFLAGS="-I$withval/include/libxml2 -I$withval/include/libxml -I$withval/include"
 )
 
@@ -74,7 +74,7 @@ AC_ARG_WITH(libxslt-prefix,
         [  --with-libxslt-prefix=[PFX]		Specify location of libxslt],
 	LIBXSLT_PREFIX=$withval
         LIBXSLT_CFLAGS="-I$withval/include/libxslt -I$withval/include/libexslt"
-        LIBXSLT_LIBS="$withval/lib/libxslt.a $withval/lib/libexslt.a -lz -lm -lpthread"
+        LIBXSLT_LIBS="$withval/lib/libxslt.so $withval/lib/libexslt.so -lz -lm -lpthread"
 )
         
 AC_ARG_WITH(libxslt-include-prefix,
@@ -84,13 +84,13 @@ AC_ARG_WITH(libxslt-include-prefix,
 
 AC_ARG_WITH(libxslt-libs-prefix,
         [  --with-libxslt-libs-prefix=[PFX]	Specify location of libxslt libs],
-        LIBXSLT_LIBS="$withval/libxslt.a $withval/libexslt.a -lz -lm -lpthread"
+        LIBXSLT_LIBS="$withval/libxslt.so $withval/libexslt.so -lz -lm -lpthread"
 )
 
 AC_ARG_WITH(libxslt-src,
         [  --with-libxslt-src=[DIR]		For libxslt thats not installed yet (sets all three above)],
         LIBXSLT_SRC="$withval"
-        LIBXSLT_LIBS="$withval/libxslt/.libs/libxslt.a $withval/libexslt/.libs/libexslt.a -lz -lm -lpthread"
+        LIBXSLT_LIBS="$withval/libxslt/.libs/libxslt.so $withval/libexslt/.libs/libexslt.so -lz -lm -lpthread"
         LIBXSLT_CFLAGS="-I$withval/libxslt -I$withval/libexslt -I$withval"
 )
 
@@ -236,7 +236,7 @@ dnl Set LIBXML libraries location
 dnl
 if test "x$LIBXML_LIBS" = "x"
 then
-	LIBXML_LIBS="${LIBXML_PREFIX}/lib/libxml2.a -lz -lm -lpthread -ldl"
+	LIBXML_LIBS="${LIBXML_PREFIX}/lib/libxml2.so -lz -lm -lpthread -ldl"
 fi
 
 dnl
@@ -252,7 +252,7 @@ dnl Set LIBXSLT libraries location
 dnl
 if test "x$LIBXSLT_LIBS" = "x"
 then
-	LIBXSLT_LIBS="${LIBXSLT_PREFIX}/lib/libxslt.a ${LIBXSLT_PREFIX}/lib/libexslt.a -lz -lm -lpthread"
+	LIBXSLT_LIBS="${LIBXSLT_PREFIX}/lib/libxslt.so ${LIBXSLT_PREFIX}/lib/libexslt.so -lz -lm -lpthread"
 fi
 
 dnl
